android-build-tools/src/apk.rs

```diff
@@ -0,0 +1,207 @@
+use crate::config::{Config, Metadata};
+use crate::error::NdkError;
+use crate::manifest::Manifest;
+use crate::ndk::{Key, Ndk};
+use crate::target::Target;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+pub struct ApkConfig {
+    pub ndk: Ndk,
+    pub build_dir: PathBuf,
+    pub assets: Option<String>,
+    pub res: Option<String>,
+    pub manifest: Manifest,
+}
+
+impl ApkConfig {
+    pub fn from_config(config: Config, metadata: Metadata) -> Self {
```

Comment:
Maybe we should pass `Config` and `Metadata` by reference to avoid cloning.

Reply:
why would you want to keep them around? `ApkConfig`'s members are pub and the `Manifest` also is pub.